src/Features/Lsif/GeneratorTest/HoverTests.vb

```diff
@@ -0,0 +1,122 @@
+
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.VisualStudio.LanguageServer.Protocol
+
+Namespace Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.UnitTests
+    <UseExportProvider>
+    Public NotInheritable Class HoverTests
+        Private Const TestProjectAssemblyName As String = "TestProject"
+
+        <Theory>
+        <InlineData("class [|C|] { string s; }", "class C")>
+        <InlineData("class C { void [|M|]() { } }", "void C.M()")>
+        <InlineData("class C { string [|s|]; }", "(field) string C.s")>
+        <InlineData("class C { void M(string [|s|]) { M(s); } }", "(parameter) string s")>
+        <InlineData("class C { void M(string s) { string [|local|] = """"; } }", "(local variable) string local")>
+        <InlineData("
+class C
+{
+    /// <summary>Doc Comment</summary>
+    void [|M|]() { }
+}",
+"void C.M()
+Doc Comment")>
+        Public Async Function TestDefinition(code As String, expectedHoverContents As String) As Task
+            Dim lsif = Await TestLsifOutput.GenerateForWorkspaceAsync(
+                TestWorkspace.CreateWorkspace(
+                    <Workspace>
+                        <Project Language="C#" AssemblyName=<%= TestProjectAssemblyName %> FilePath="Z:\TestProject.csproj" CommonReferences="true">
+                            <Document Name="A.cs" FilePath="Z:\A.cs">
+                                <%= code %>
+                            </Document>
+                        </Project>
+                    </Workspace>))
+
+            Dim rangeVertex = Await lsif.GetSelectedRangeAsync()
+            Dim resultSetVertex = lsif.GetLinkedVertices(Of Graph.ResultSet)(rangeVertex, "next").Single()
+            Dim hoverVertex = lsif.GetLinkedVertices(Of Graph.HoverResult)(resultSetVertex, Methods.TextDocumentHoverName).SingleOrDefault()
+            Dim hoverMarkupContent = DirectCast(hoverVertex.Result.Contents.Value, MarkupContent)
+
+            Assert.Equal(MarkupKind.PlainText, hoverMarkupContent.Kind)
+            Assert.Equal(expectedHoverContents + Environment.NewLine, hoverMarkupContent.Value)
+        End Function
+
+        <Theory>
+        <InlineData("class C { [|string|] s; }", "class System.String")>
+        <InlineData("class C { void M() { [|M|](); } }", "void C.M()")>
+        <InlineData("class C { void M(string s) { M([|s|]); } }", "(parameter) string s")>
+        <InlineData("class C { void M(string s) { string local = """"; M([|local|]); } }", "(local variable) string local")>
+        <InlineData("using [|S|] = System.String;", "class System.String")>
+        <InlineData("class C { [|global|]::System.String s; }", "<global namespace>")>
+        <InlineData("
+class C
+{
+    /// <see cref=""C.[|M|]()"" />
+    void M() { }
+}",
+"void C.M()")>
+        Public Async Function TestReference(code As String, expectedHoverContents As String) As Task
```

Comment:
💡 Rather than use a parameter for the expected result, you can use a switch expression. This allows the test inputs to focus on inputs.

```
var expectedHoverContents = code switch
{
  "class C { [|string|] s; }" => "class System.String",
  // ...
  _ => throw new NotSupportedException("Unrecognized input."),
};
```

Reply:
I have followed the pattern used in most existing tests, will consider your approach going forward.